lib/json/json.go

```diff
@@ -200,8 +199,15 @@ func encode(thread *starlark.Thread, b *starlark.Builtin, args starlark.Tuple, k
 			sort.Strings(names)
 			for i, name := range names {
 				v, err := x.Attr(name)
-				if err != nil || v == nil {
-					log.Fatalf("internal error: dir(%s) includes %q but value has no .%s field", x.Type(), name, name)
+				if err != nil {
+					return fmt.Errorf("reading attribute %s.%q failed: %w", x.Type(), name, err)
```

Comment:
Let's use this wording:  "cannot access attribute %s.%s: %w", Type(), name, err
And: "missing attribute %s.%s (despite %q appearing in dir())", Type(), name, name

Quotation seems unnecessary: yes, it is technically possible for AttrNames to contains arbitrary strings but in practice they are always identifiers.

Reply:
done